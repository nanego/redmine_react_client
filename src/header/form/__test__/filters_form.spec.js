import React from 'react';
import FiltersForm from '../filters_form';
import renderer from 'react-test-renderer';

let applyFiltersChanges = function(){};
let updateSelectedFilters = function(){};
let searchInputValue = function(){};
let handleSearchInputChange  = function(){};

it('should render FiltersForm with empty filters', () => {
  let current_filters = {};
  let selected_filters = {};
  let areFiltersDirty = false;
  const component = renderer.create(
      <FiltersForm current_filters={current_filters}
                   selected_filters={selected_filters}
                   applyFiltersChanges={applyFiltersChanges}
                   updateSelectedFilters={updateSelectedFilters}
                   searchValue={searchInputValue}
                   updateSearchValue={handleSearchInputChange}
                   areFiltersDirty={areFiltersDirty}
      />
  );
  let select = component.toJSON();
  expect(select).toMatchSnapshot();
});

it('should render FiltersForm with NOT empty filters', () => {
  let current_filters = { "projects": { "operator": "=", "value": 1 }, "trackers": { "operator": "=", "value": 2 }, "status": { "operator": "=", "value": 2 }, "watched": { "operator": "=", "value": "true" }, "assigned_to": { "operator": "=", "value": 1 }, "updated_at": { "operator": ">", "value": "12/12/2017" }, "text": "test" };
  let selected_filters = { "text": "" };
  let areFiltersDirty = true;
  const component = renderer.create(
      <FiltersForm current_filters={current_filters}
                   selected_filters={selected_filters}
                   applyFiltersChanges={applyFiltersChanges}
                   updateSelectedFilters={updateSelectedFilters}
                   searchValue={searchInputValue}
                   updateSearchValue={handleSearchInputChange}
                   areFiltersDirty={areFiltersDirty}
      />
  );
  let select = component.toJSON();
  expect(select).toMatchSnapshot();
});
