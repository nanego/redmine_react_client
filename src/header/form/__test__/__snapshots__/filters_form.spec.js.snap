exports[`test should render FiltersForm 1`] = `
<form
  className="ui form filters_form">
  <div
    className="inline field">
    <label>
      Projets
    </label>
    <div
      aria-busy={false}
      aria-disabled={undefined}
      aria-expanded={false}
      className="ui search selection left pointing link item dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="combobox"
      tabIndex={undefined}>
      <select
        aria-hidden="true"
        multiple={undefined}
        name={undefined}
        type="hidden"
        value="">
        <option
          value="" />
        <option
          value={1}>
          eCookbook
        </option>
        <option
          value={5}>
          Private child of eCookbook
        </option>
        <option
          value={6}>
          Child of private child
        </option>
        <option
          value={3}>
          eCookbook Subproject 1
        </option>
        <option
          value={4}>
          eCookbook Subproject 2
        </option>
        <option
          value={2}>
          OnlineStore
        </option>
      </select>
      <input
        aria-autocomplete="list"
        autoComplete="off"
        className="search"
        name="-search"
        onChange={[Function]}
        style={
          Object {
            "width": undefined,
          }
        }
        tabIndex={0}
        type="text"
        value="" />
      <div
        className="default text">
        Projets
      </div>
      <i
        aria-hidden="true"
        className="dropdown icon" />
      <div
        aria-multiselectable={undefined}
        className="menu transition"
        role="listbox">
        <div
          aria-checked={false}
          aria-disabled={undefined}
          aria-selected={true}
          className="selected item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }>
          <span
            className="text">
            eCookbook
          </span>
        </div>
        <div
          aria-checked={false}
          aria-disabled={undefined}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }>
          <span
            className="text">
            Private child of eCookbook
          </span>
        </div>
        <div
          aria-checked={false}
          aria-disabled={undefined}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }>
          <span
            className="text">
            Child of private child
          </span>
        </div>
        <div
          aria-checked={false}
          aria-disabled={undefined}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }>
          <span
            className="text">
            eCookbook Subproject 1
          </span>
        </div>
        <div
          aria-checked={false}
          aria-disabled={undefined}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }>
          <span
            className="text">
            eCookbook Subproject 2
          </span>
        </div>
        <div
          aria-checked={false}
          aria-disabled={undefined}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }>
          <span
            className="text">
            OnlineStore
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="inline field">
    <label>
      Trackers
    </label>
    <div
      aria-busy={false}
      aria-disabled={undefined}
      aria-expanded={false}
      className="ui search selection left pointing link item dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="combobox"
      tabIndex={undefined}>
      <select
        aria-hidden="true"
        multiple={undefined}
        name={undefined}
        type="hidden"
        value="">
        <option
          value="" />
        <option
          value={1}>
          Bug
        </option>
        <option
          value={2}>
          Feature request
        </option>
        <option
          value={3}>
          Support request
        </option>
      </select>
      <input
        aria-autocomplete="list"
        autoComplete="off"
        className="search"
        name="-search"
        onChange={[Function]}
        style={
          Object {
            "width": undefined,
          }
        }
        tabIndex={0}
        type="text"
        value="" />
      <div
        className="default text">
        Trackers
      </div>
      <i
        aria-hidden="true"
        className="dropdown icon" />
      <div
        aria-multiselectable={undefined}
        className="menu transition"
        role="listbox">
        <div
          aria-checked={false}
          aria-disabled={undefined}
          aria-selected={true}
          className="selected item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }>
          <span
            className="text">
            Bug
          </span>
        </div>
        <div
          aria-checked={false}
          aria-disabled={undefined}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }>
          <span
            className="text">
            Feature request
          </span>
        </div>
        <div
          aria-checked={false}
          aria-disabled={undefined}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }>
          <span
            className="text">
            Support request
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="inline field">
    <label>
      Statuts
    </label>
    <div
      aria-busy={false}
      aria-disabled={undefined}
      aria-expanded={false}
      className="ui search selection left pointing link item dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="combobox"
      tabIndex={undefined}>
      <select
        aria-hidden="true"
        multiple={undefined}
        name={undefined}
        type="hidden"
        value="">
        <option
          value="" />
        <option
          value={1}>
          New
        </option>
        <option
          value={2}>
          In Progress
        </option>
        <option
          value={3}>
          Done
        </option>
        <option
          value={4}>
          Feedback
        </option>
        <option
          value={5}>
          Closed
        </option>
        <option
          value={6}>
          Rejected
        </option>
      </select>
      <input
        aria-autocomplete="list"
        autoComplete="off"
        className="search"
        name="-search"
        onChange={[Function]}
        style={
          Object {
            "width": undefined,
          }
        }
        tabIndex={0}
        type="text"
        value="" />
      <div
        className="default text">
        Issue Statuses
      </div>
      <i
        aria-hidden="true"
        className="dropdown icon" />
      <div
        aria-multiselectable={undefined}
        className="menu transition"
        role="listbox">
        <div
          aria-checked={false}
          aria-disabled={undefined}
          aria-selected={true}
          className="selected item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }>
          <span
            className="text">
            New
          </span>
        </div>
        <div
          aria-checked={false}
          aria-disabled={undefined}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }>
          <span
            className="text">
            In Progress
          </span>
        </div>
        <div
          aria-checked={false}
          aria-disabled={undefined}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }>
          <span
            className="text">
            Done
          </span>
        </div>
        <div
          aria-checked={false}
          aria-disabled={undefined}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }>
          <span
            className="text">
            Feedback
          </span>
        </div>
        <div
          aria-checked={false}
          aria-disabled={undefined}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }>
          <span
            className="text">
            Closed
          </span>
        </div>
        <div
          aria-checked={false}
          aria-disabled={undefined}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }>
          <span
            className="text">
            Rejected
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="inline field">
    <label>
      Observateur
    </label>
    <div
      aria-busy={undefined}
      aria-disabled={undefined}
      aria-expanded={false}
      className="ui search selection left pointing link item dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="combobox"
      tabIndex={undefined}>
      <select
        aria-hidden="true"
        multiple={undefined}
        name={undefined}
        type="hidden"
        value="">
        <option
          value="" />
        <option
          value="">
          
        </option>
        <option
          value="true">
          Oui
        </option>
        <option
          value="false">
          Non
        </option>
      </select>
      <input
        aria-autocomplete="list"
        autoComplete="off"
        className="search"
        name="-search"
        onChange={[Function]}
        style={
          Object {
            "width": undefined,
          }
        }
        tabIndex={0}
        type="text"
        value="" />
      <div
        className="text" />
      <i
        aria-hidden="true"
        className="dropdown icon" />
      <div
        aria-multiselectable={undefined}
        className="menu transition"
        role="listbox">
        <div
          aria-checked={true}
          aria-disabled={undefined}
          aria-selected={true}
          className="active selected item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }>
          <span
            className="text">
            
          </span>
        </div>
        <div
          aria-checked={false}
          aria-disabled={undefined}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }>
          <span
            className="text">
            Oui
          </span>
        </div>
        <div
          aria-checked={false}
          aria-disabled={undefined}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }>
          <span
            className="text">
            Non
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="inline field">
    <label>
      Date de mise à jour
    </label>
    <div
      className="ui left action input">
      <div
        aria-busy={undefined}
        aria-disabled={undefined}
        aria-expanded={false}
        aria-multiselectable={undefined}
        className="ui compact selection dropdown"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        role="listbox"
        tabIndex={0}>
        <select
          aria-hidden="true"
          multiple={undefined}
          name={undefined}
          type="hidden"
          value="=">
          <option
            value="" />
          <option
            value="=">
            =
          </option>
          <option
            value="<">
            <
          </option>
          <option
            value=">">
            >
          </option>
        </select>
        <div
          className="text">
          =
        </div>
        <i
          aria-hidden="true"
          className="dropdown icon" />
        <div
          className="menu transition">
          <div
            aria-checked={true}
            aria-disabled={undefined}
            aria-selected={true}
            className="active selected item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }>
            <span
              className="text">
              =
            </span>
          </div>
          <div
            aria-checked={false}
            aria-disabled={undefined}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }>
            <span
              className="text">
              <
            </span>
          </div>
          <div
            aria-checked={false}
            aria-disabled={undefined}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }>
            <span
              className="text">
              >
            </span>
          </div>
        </div>
      </div>
      <input
        onChange={[Function]}
        placeholder="DD/MM/YYYY"
        type="text"
        value={undefined} />
    </div>
  </div>
  <div
    className="inline field"
    id="form_content_field">
    <label>
      Contient
    </label>
    <div
      className="ui input">
      <input
        onChange={[Function]}
        placeholder="contenu recherché"
        type="text"
        value={undefined} />
    </div>
  </div>
  <div
    className="ui divider" />
  <button
    className="ui blue button"
    name="apply_filters"
    onClick={[Function]}
    tabIndex={undefined}
    type="submit">
    Appliquer
  </button>
  <button
    className="ui button"
    name="reset_filters"
    onClick={[Function]}
    tabIndex={undefined}>
    Effacer
  </button>
</form>
`;
